src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -508,12 +508,99 @@ namespace AppInstaller::CLI::Workflow
                 return result;
             }
         };
+
+        struct MarketFilter : public details::FilterField
+        {
+            MarketFilter(Manifest::string_t market) : details::FilterField("Market"), m_market(market)
+            {
+                AICLI_LOG(CLI, Verbose, << "Market Comparator created with market: " << m_market);
+            }
+
+            static std::unique_ptr<MarketFilter> Create()
+            {
+                return std::make_unique<MarketFilter>(GetCurrentMarket());
+            }
+
+            InapplicabilityFlags IsApplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                // If both allowed and excluded lists are provided, we only need to check the allowed markets.
+                if (!installer.Markets.AllowedMarkets.empty())
+                {
+                    // Inapplicable if NOT found
+                    if (installer.Markets.AllowedMarkets.end() == std::find(installer.Markets.AllowedMarkets.begin(), installer.Markets.AllowedMarkets.end(), m_market))
+                    {
+                        return InapplicabilityFlags::Market;
+                    }
+                }
+                else if (!installer.Markets.ExcludedMarkets.empty())
+                {
+                    // Inapplicable if found
+                    if (installer.Markets.ExcludedMarkets.end() != std::find(installer.Markets.ExcludedMarkets.begin(), installer.Markets.ExcludedMarkets.end(), m_market))
+                    {
+                        return InapplicabilityFlags::Market;
+                    }
+                }
+
+                return InapplicabilityFlags::None;
+            }
+
+            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                std::string result = "Current market '" + m_market + "' does not match installer markets." +
+                    " Allowed markets: " + GetMarketsListAsString(installer.Markets.AllowedMarkets) +
+                    " Excluded markets: " + GetMarketsListAsString(installer.Markets.ExcludedMarkets);
+                return result;
+            }
+
+        private:
+            Manifest::string_t m_market;
+
+            static Manifest::string_t GetCurrentMarket()
+            {
+                // TODO: Manifest uses 2-letter codes (ISO 3166?) but GetUserDefaultGeoName can return
+                // either a 2-letter ISO-3166 code or a numeric UN M.49 code.
+                auto geoNameSize = GetUserDefaultGeoName(nullptr, 0);

```

Comment:
I'm not actually 100% sure whether this matches what was intended with this manifest field, but it should be easy to change if it's not :)
I'm somewhat worried about the fact that this function can return 3-digit numeric codes, which are not supported in the manifest. I couldn't find anything about when would it return ISO 3166 or UN M.49.

Reply:
I would hope that the manifest field and the REST market were considered at the same time and should be the same format.

Reply:
Changed to use the existing code which uses the 2 letter format.